lms/djangoapps/teams/csv.py

```diff
@@ -292,18 +300,17 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                     return False
                 else:
                     self.user_ids_by_teamset_id[teamset_id].add(user.id)
-                    continue
+                    #continue
             max_team_size = self.course.teams_configuration.default_max_team_size
-            if max_team_size is not None and team.users.count() >= max_team_size:
-                if self.add_error_and_check_if_max_exceeded('Team ' + team.team_id + ' is full.'):
+            key = '{}.{}'.format(teamset_id, team_name)
+            if key not in self.team_name_to_proposed_membership:
+                self.team_name_to_proposed_membership[key] = initial_membership_count + 1
+            else:
+                self.team_name_to_proposed_membership[key] += 1
+            if max_team_size is not None and self.team_name_to_proposed_membership[key] >= max_team_size:
+                if self.add_error_and_check_if_max_exceeded('Team ' + team_name + ' is full.'):
```

Comment:
The team size check feels like it should be a separate function to follow better single-responsibility design principles.

Reply:
I think they are pretty closely tied. In addition, we are looping over a list of teamsets. I want to avoid extra loops.